go/extra/ba/README.md

```diff
@@ -0,0 +1,61 @@
+# Benchmark analysis
+
+This tool compares benchmark results generated by `oasis-test-runner` and
+corresponding `oasis-node` workers.
+
+## Usage
+
+Currently, `cmp` operation is implemented which compares the benchmark results
+of the last (`source`) and pre-last (`target`) test batches (also called
+*instances*):
+```bash
+$ ba/ba cmp \
+  --metrics.address http://localhost:9090
+```
+
+By default results of all `oasis-test-runner` tests and metrics will be
+compared. If you want to compare specific metric(s) and specific test(s),
+provide `-m` and `-t` flags respectively:
+```bash
+$ ba/ba cmp \
+  -m time \
+  -t multiple-runtimes \
+  --metrics.address http://localhost:9090
+```
+
+Comparison is done by considering *thresholds* for each metric. Currently,
+we support `avg_ratio` and `max_ratio` which correspond to the average and
+maximum ratio of metric values of all test runs in the benchmark batches. For
+each ratio, you can set the `max_threshold` and `min_threshold`. The former
+requires that the ratio should not be exceeded and the latter that the provided
+threshold must be reached. If not, `ba` exits with error code. Thresholds can
+be using the flag `--{min|max}_threshold.<metric name>.{avg|max}_ratio`.
+
+For example:
+```bash
+$ ba/ba cmp \
+  -m time \
+  -t multiple-runtimes \
+  --max_threshold.time.avg_ratio 1.1 \
+  --metrics.address http://localhost:9090
+```
+
+will require that the average duration of the test in the last benchmark batch
+should be at most 10\% slower than from the pre-last benchmark batch.
+
+If you are developing or improving a feature in a separate git branch, you will
+want to perform benchmarks and compare the results of your branch to the ones
+from the master branch. In this case, you can set the git branch of the source
+and target batch. `ba` will compare the last benchmark result from each git
+branch:
```

Comment:
Note should be added that this requires that you first run benchmarks master and your branches and report results to the same (probably local) prometheus instance